src/diagrams/common/common.ts

```diff
@@ -0,0 +1,147 @@
+  if (!s) return [''];
+  const str = breakToPlaceholder(s).replace(/\\n/g, '#br#');
+  return str.split('#br#');
+};
+
+/**
+ * Removes script tags from a text
+ *
+ * @param {string} txt The text to sanitize
+ * @returns {string} The safer text
+ */
+export const removeScript = (txt: string): string => {
+  return DOMPurify.sanitize(txt);
+};
+
+const sanitizeMore = (text: string, config: MermaidConfig) => {
+  // TODO Q: Should this check really be here? Feels like we should be sanitizing it regardless.
+  if (config.flowchart?.htmlLabels !== false) {
+    const level = config.securityLevel;
+    if (level === 'antiscript' || level === 'strict') {
+      text = removeScript(text);
+    } else if (level !== 'loose') {
+      text = breakToPlaceholder(text);
+      text = text.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+      text = text.replace(/=/g, '&equals;');
+      text = placeholderToBreak(text);
+    }
+  }
+  return text;
+};
+
+export const sanitizeText = (text: string, config: MermaidConfig): string => {
+  if (!text) return text;
+  if (config.dompurifyConfig) {
+    text = DOMPurify.sanitize(sanitizeMore(text, config), config.dompurifyConfig).toString();
+  } else {
+    text = DOMPurify.sanitize(sanitizeMore(text, config));
+  }
+  return text;
+};
+
+export const sanitizeTextOrArray = (
+  a: string | string[] | string[][],
+  config: MermaidConfig
+): string | string[] => {
+  if (typeof a === 'string') return sanitizeText(a, config);
+  // TODO Q: Do we need flat?
+  return a.flat().map((x: string) => sanitizeText(x, config));
+};
+
+export const lineBreakRegex = /<br\s*\/?>/gi;
+
+/**
+ * Whether or not a text has any linebreaks
+ *
+ * @param {string} text The text to test
+ * @returns {boolean} Whether or not the text has breaks
+ */
+export const hasBreaks = (text: string): boolean => {
+  return lineBreakRegex.test(text);
+};
+
+/**
+ * Splits on <br> tags
+ *
+ * @param {string} text Text to split
+ * @returns {string[]} List of lines as strings
+ */
+export const splitBreaks = (text: string): string[] => {
+  return text.split(lineBreakRegex);
+};
+
+/**
+ * Converts placeholders to linebreaks in HTML
+ *
+ * @param {string} s HTML with placeholders
+ * @returns {string} HTML with breaks instead of placeholders
+ */
+const placeholderToBreak = (s: string): string => {
+  return s.replace(/#br#/g, '<br/>');
+};
+
+/**
+ * Opposite of `placeholderToBreak`, converts breaks to placeholders
+ *
+ * @param {string} s HTML string
+ * @returns {string} String with placeholders
+ */
+const breakToPlaceholder = (s: string): string => {
+  return s.replace(lineBreakRegex, '#br#');
+};
+
+/**
+ * Gets the current URL
+ *
+ * @param {boolean} useAbsolute Whether to return the absolute URL or not
+ * @returns {string} The current URL
+ */
+const getUrl = (useAbsolute: boolean): string => {
+  let url = '';
+  // TODO Q: If useAbsolute if false, empty string is returned. Bug?
+  if (useAbsolute) {
```

Comment:
@knsv

Reply:
It should be like this, it is about references to the symbols at the end of the lines. With use absolute you have the full url and without useAbsolute it is ok with an empty string